docs/calyx-py.md

```diff
@@ -1,17 +1,120 @@
 # Emitting Calyx from Python
 
-Our frontends are written in Python3 and make use of the `calyx` library to
+Our frontends are written in Python3 and make use of the `calyx` builder library to
 generate their code.
 
+## Installation
+
 To install the library, run the following from the repository root (requires
 [flit][] installation):
+
 ```
 cd calyx-py && flit install -s
 ```
 
-The library provides an example:
+## Using the `calyx` builder
+
+The `calyx` library provides a builder to generate Calyx code. Let's walk through the file `calyx-py/test/builder_example.py`. This Calyx program initializes two registers with the numbers 1 and 41, adds them together, and stores the result in a register.
+
+We use the `Builder` object to construct our Calyx program `prog`. We can define components with `prog.component`. Here's a defininition of a `main` component with a 32-bit input and output:
```

Comment:
Maybe it's just me, but the example plus the writing makes it look like the input needs be named "in" and the output "out". Can you tweak the writing, e.g. "32-bit input called `in` and ..."?